File: simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/core/navigation/BaseFragment.java
package com.zhuinden.navigationexamplefrag.core.navigation;

import android.support.v4.app.Fragment;

/**
 * Created by Owner on 2017. 06. 29..
 */

public class BaseFragment extends Fragment {
    public final <T extends BaseKey> T getKey() {
        return getArguments() != null ? getArguments().<T>getParcelable("KEY") : null;
    }
}

File: simple-stack-example-basic-kotlin-fragment/src/main/java/com/zhuinden/simplestackexamplekotlinfragment/core/navigation/BaseFragment.kt
package com.zhuinden.simplestackexamplekotlinfragment.core.navigation

import android.support.v4.app.Fragment
import com.zhuinden.utils.requireArguments

/**
 * Created by Owner on 2017.11.13.
 */

open class BaseFragment : Fragment() {
    fun <T : BaseKey> getKey(): T = requireArguments.getParcelable<T>("KEY")!!
}
